ert_gui/tools/plot/plot_window.py

```diff
@@ -92,7 +92,13 @@ def currentPlotChanged(self):
                     and plot_widget._plotter.dimensionality == key_def["dimensionality"]:
                 self._updateCustomizer(plot_widget)
                 cases = self._case_selection_widget.getPlotCaseNames()
-                case_to_data_map = {case: self._api.data_for_key(case, key)[key] for case in cases}
+                def get_key(df):
+                    if not df.empty:
+                        return df[key]
+                    import pandas as pd
+                    return pd.DataFrame()
+
+                case_to_data_map = {case: get_key(self._api.data_for_key(case, key)) for case in cases}
```

Comment:
This is a hack to confirm my suspicions! I suggest we change the API such that data_for_key(case, key) returns a `pandas.DataFrame` withouth a `pandas.MultiIndex` with key and index, and rather a `pandas.DataFrame` with only `pandas.Index` of the indexes.

Reply:
Backside of doing that is that we actually change the API, which means that the new client in the storage code also will need to be changed before that is merged.

Reply:
Update: suggestion made in newest commit

Reply:
I support the idea of the changes, but I won't verify that the bugs are fixed tonight ;)